cmd/cmdoptions/scenariorunner.go

```diff
@@ -0,0 +1,117 @@
+package cmdoptions
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/spf13/pflag"
+	"github.com/temporalio/omes/loadgen"
+	"go.temporal.io/sdk/client"
+	"go.uber.org/zap"
+)
+
+type ScenarioRunner struct {
+	Logger          *zap.SugaredLogger
+	Scenario        string
+	RunID           string
+	Iterations      int
+	Duration        time.Duration
+	MaxConcurrent   int
+	ScenarioOptions []string
+	ConnectTimeout  time.Duration
+	ClientOptions   ClientOptions
+	MetricsOptions  MetricsOptions
+	LoggingOptions  LoggingOptions
+}
+
+// AddCLIFlags defines CLI flags that will hydrate the ScenarioRunner struct.
+func (r *ScenarioRunner) AddCLIFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&r.Scenario, "scenario", "", "Scenario name to run")
+	fs.StringVar(&r.RunID, "run-id", "", "Run ID for this run")
+	fs.IntVar(&r.Iterations, "iterations", 0, "Override default iterations for the scenario (cannot be provided with duration)")
+	fs.DurationVar(&r.Duration, "duration", 0, "Override duration for the scenario (cannot be provided with iteration)")
+	fs.IntVar(&r.MaxConcurrent, "max-concurrent", 0, "Override max-concurrent for the scenario")
+	fs.StringSliceVar(&r.ScenarioOptions, "option", nil, "Additional options for the scenario, in key=value format")
+	fs.DurationVar(&r.ConnectTimeout, "connect-timeout", 0, "Duration to try to connect to server before failing")
+	r.ClientOptions.AddCLIFlags(fs)
+	r.MetricsOptions.AddCLIFlags(fs, "")
+	r.LoggingOptions.AddCLIFlags(fs)
+}
+
+// Run runs the scenario.
+func (r *ScenarioRunner) Run(ctx context.Context) error {
```

Comment:
I know coming from other languages what's happening here may _look_ like a lot of logic, but it's not really. Just `loadgen.GetScenario` + `loadgen.ScenarioInfo` creation + `scenario.Executor.Run`. There's not need to pull all of this out from where it was and try to reuse it.

Can you show the code that will use this so we can discuss alternatives?

Reply:
> Can you show the code that will use this so we can discuss alternatives?

Please see the PR introduction.

Reply:
Let me make an alternative version of the `sass-cicd` caller that does not rely on exposing `ScenarioRunner` in omes, to see what it looks like.

Reply:
I agree with you @cretz. Duplicating the 50 lines or so between omes CLI and the cicd runner is the right decision, rather than exposing new abstractions in omes that there's not an urgent need for. I've done that here https://github.com/dandavison/temporalio-saas-cicd/blob/sdk-1421-omes/oss-cicd/e2e/workflows/test_workload/activities/omes.go and reverted that part of this PR.